### PR TITLE
[Feature](bangc-ops): Synchronize common code include core and gtest etc.

### PR DESCRIPTION
**Pull request by ZhangLearning**
Thanks for your contribution and we appreciate it a lot. 

## 1. Motivation

Synchronize common code include core and gtest code.

## 2. Modification

bangc-ops/core/cnlog.cpp
bangc-ops/core/cnlog.h
bangc-ops/core/context.cpp
bangc-ops/core/context.h
bangc-ops/core/gen_case.cpp
bangc-ops/core/gen_case.h
bangc-ops/core/logging.cpp
bangc-ops/core/logging.h
bangc-ops/core/macros.h
bangc-ops/core/mlu_env.cpp
bangc-ops/core/mlu_env.h
bangc-ops/core/platform/posix/env_time.cpp
bangc-ops/core/preprocessor.h
bangc-ops/core/runtime/device.h
bangc-ops/core/tensor.cpp
bangc-ops/core/tensor.h
bangc-ops/core/tool.cpp
bangc-ops/core/tool.h
bangc-ops/core/type.cpp
bangc-ops/core/type.h
bangc-ops/core/util.cpp
bangc-ops/kernels/kernel.h
bangc-ops/kernels/tensor_stride_process/tensor_stride_process.cpp
bangc-ops/mlu_op.h
bangc-ops/test/mlu_op_gtest/googletest/include/gtest/gtest-message.h
bangc-ops/test/mlu_op_gtest/googletest/include/gtest/gtest-param-test.h
bangc-ops/test/mlu_op_gtest/googletest/include/gtest/gtest-spi.h
bangc-ops/test/mlu_op_gtest/googletest/include/gtest/gtest-test-part.h
bangc-ops/test/mlu_op_gtest/googletest/include/gtest/gtest-typed-test.h
bangc-ops/test/mlu_op_gtest/googletest/include/gtest/gtest.h
bangc-ops/test/mlu_op_gtest/googletest/include/gtest/internal/gtest-death-test-internal.h
bangc-ops/test/mlu_op_gtest/googletest/include/gtest/internal/gtest-filepath.h
bangc-ops/test/mlu_op_gtest/googletest/include/gtest/internal/gtest-internal.h
bangc-ops/test/mlu_op_gtest/googletest/include/gtest/internal/gtest-linked_ptr.h
bangc-ops/test/mlu_op_gtest/googletest/include/gtest/internal/gtest-param-util.h
bangc-ops/test/mlu_op_gtest/googletest/include/gtest/internal/gtest-port.h
bangc-ops/test/mlu_op_gtest/googletest/include/gtest/internal/gtest-tuple.h
bangc-ops/test/mlu_op_gtest/googletest/include/gtest/internal/gtest-type-util.h
bangc-ops/test/mlu_op_gtest/googletest/include/gtest/mlu_op_test_case.h
bangc-ops/test/mlu_op_gtest/googletest/src/gtest.cc
bangc-ops/test/mlu_op_gtest/include/tools.h
bangc-ops/test/mlu_op_gtest/pb_gtest/include/case_collector.h
bangc-ops/test/mlu_op_gtest/pb_gtest/include/evaluator.h
bangc-ops/test/mlu_op_gtest/pb_gtest/include/executor.h
bangc-ops/test/mlu_op_gtest/pb_gtest/include/memory_pool.h
bangc-ops/test/mlu_op_gtest/pb_gtest/include/parser.h
bangc-ops/test/mlu_op_gtest/pb_gtest/include/pb_test_tools.h
bangc-ops/test/mlu_op_gtest/pb_gtest/include/runtime.h
bangc-ops/test/mlu_op_gtest/pb_gtest/include/variable.h
bangc-ops/test/mlu_op_gtest/pb_gtest/mlu_op_test_proto
bangc-ops/test/mlu_op_gtest/pb_gtest/src/evaluator.cpp
bangc-ops/test/mlu_op_gtest/pb_gtest/src/executor.cpp
bangc-ops/test/mlu_op_gtest/pb_gtest/src/gtest/accuracy_test.cpp
bangc-ops/test/mlu_op_gtest/pb_gtest/src/gtest/accuracy_test.h
bangc-ops/test/mlu_op_gtest/pb_gtest/src/gtest/mlu_op_gtest.cpp
bangc-ops/test/mlu_op_gtest/pb_gtest/src/gtest/mlu_op_gtest.h
bangc-ops/test/mlu_op_gtest/pb_gtest/src/gtest/mlu_op_gtest_event_listener.h
bangc-ops/test/mlu_op_gtest/pb_gtest/src/gtest/mlu_op_test_result_printer.cpp
bangc-ops/test/mlu_op_gtest/pb_gtest/src/gtest/mlu_op_test_result_printer.h
bangc-ops/test/mlu_op_gtest/pb_gtest/src/gtest/register_op.py
bangc-ops/test/mlu_op_gtest/pb_gtest/src/gtest/test_env.cpp
bangc-ops/test/mlu_op_gtest/pb_gtest/src/gtest/test_env.h
bangc-ops/test/mlu_op_gtest/pb_gtest/src/internal_kernel/fill_ram/fill_ram.h
bangc-ops/test/mlu_op_gtest/pb_gtest/src/internal_kernel/fill_ram/fill_ram_device.mlu
bangc-ops/test/mlu_op_gtest/pb_gtest/src/internal_perf.cpp
bangc-ops/test/mlu_op_gtest/pb_gtest/src/kl_test.cpp
bangc-ops/test/mlu_op_gtest/pb_gtest/src/main.cpp
bangc-ops/test/mlu_op_gtest/pb_gtest/src/parser.cpp
bangc-ops/test/mlu_op_gtest/pb_gtest/src/pb_test_tools.cpp
bangc-ops/test/mlu_op_gtest/pb_gtest/src/runtime.cpp
bangc-ops/test/mlu_op_gtest/pb_gtest/src/stats_test.h
bangc-ops/test/mlu_op_gtest/pb_gtest/src/zero_element.h
bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/active_rotated_filter_forward/active_rotated_filter_forward.cpp
bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/ball_query/ball_query.cpp
bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/three_nn_forward/three_nn_forward.cpp
bangc-ops/test/mlu_op_gtest/pb_gtest/tests/mlu_op_test.h
bangc-ops/test/mlu_op_gtest/tools/generate_case.py
bangc-ops/test/mlu_op_gtest/tools/gtest_style.py
bangc-ops/test/mlu_op_gtest/tools/pb2prototxt.cpp
bangc-ops/test/mlu_op_gtest/tools/prototxt2pb.cpp
tools/perf_analyse/utils.py

## 3. Test Report

CPU case:
Platform：MLU370
[----------] Global test environment tear-down
[ SUMMARY  ] Total 94 cases of 57 op(s).
ALL PASSED.
[==========] 94 test cases from 57 test suites ran. (12166 ms total)
[  PASSED  ] 94 test cases.

Platform：MLU590
[----------] Global test environment tear-down
[ SUMMARY  ] Total 94 cases of 57 op(s).
ALL PASSED.
[==========] 94 test cases from 57 test suites ran. (11889 ms total)
[  PASSED  ] 94 test cases.

release_temp + release_test:

Platform：MLU370-X8
[----------] Global test environment tear-down
[ SUMMARY  ] Total 15019 cases of 51 op(s).
ALL PASSED.
[==========] 15019 test cases from 51 test suites ran. (4309665 ms total)
[  PASSED  ] 15019 test cases.

Platform：MLU370-X4
[----------] Global test environment tear-down
[ SUMMARY  ] Total 15019 cases of 51 op(s).
ALL PASSED.
[==========] 15019 test cases from 51 test suites ran. (14173222 ms total)
[  PASSED  ] 15019 test cases.

Platform：MLU590-H8
[----------] Global test environment tear-down
[ SUMMARY  ] Total 14946 cases of 51 op(s).
ALL PASSED.
[==========] 14946 test cases from 51 test suites ran. (4961810 ms total)
[  PASSED  ] 14946 test cases.

### 3.4 Summary Analysis

All test case PASS on mlu590 and mlu370.

### CODE REVIEW

File: bangc-ops/mlu_op.h [hunk 1]
```diff
@@ -1872,6 +2037,48 @@ mluOpGetTensorDescriptorEx(const mluOpTensorDescriptor_t desc,
...
                            int dimStride[]);
 
+// Group:Common Interface:Tensor
+/*!
+ *  @brief Retrieves a tensor descriptor \b desc that is previously created with the
+ *  ::mluOpCreateTensorDescriptor and sets the information about the dimensions, data type,
+ *  stride and layout of input tensor with ::mluOpSetTensorDescriptorEx_v2.
+ *
+ *  @param[in] desc
+ *    Input. The descriptor of the input tensor. For detailed information,
```
Comment (Wickyzheng) [inline bangc-ops/mlu_op.h:2047]: mluops的规范是和@对齐。你看下https://wiki.cambricon.com/pages/viewpage.action?pageId=255658083

File: bangc-ops/mlu_op.h [hunk 2]
```diff
@@ -1818,6 +1944,45 @@ mluOpStatus_t MLUOP_WIN_API
...
+ *    see ::mluOpTensorDescriptor_t.
+ *  @param[out] layout
+ *    Output. Pointer to the host memory that holds information about the layout of the input tensor.
+ *  For detailed information, see ::mluOpTensorLayout_t.
+ *  @param[out] dtype
+ *    Output. Pointer to the host memory that holds information about the data type of the input tensor.
+ *  For detailed information, see ::mluOpDataType_t.
+ *  @param[out] dimNb
+ *    Output. Pointer to the host memory that holds information about the dimension of input tensor.
```
Comment (Wickyzheng) [inline bangc-ops/mlu_op.h:1963]: mluops的格式中没有Output这种字眼了吧。参考这个wiki 统一改下吧  还有一些对齐的格式。https://wiki.cambricon.com/pages/viewpage.action?pageId=255658083

**Review by Wickyzheng — APPROVED** [review]

**Review by defei-coder — APPROVED** [review]

File: bangc-ops/test/mlu_op_gtest/pb_gtest/src/gtest/mlu_op_gtest.cpp [hunk 3]
```diff
@@ -667,4 +682,16 @@ void TestSuite::recordXml(mluoptest::EvaluateResult &er) {
...
+  // record gtest internal perf
+  if (global_var.enable_gtest_internal_perf) {
+    for (auto &it : er.gtest.time_costs_ms) {
+      this->RecordProperty("gtest_time_cost_ms_" + std::get<0>(it),
+                           std::to_string(std::get<1>(it)));
+    }
+    this->RecordProperty("gtest_case_file_mb",
+                         std::to_string(er.gtest.parsed_file_size/1024./1024.));
+  }
+#endif
```
Comment (Unireverse) [inline bangc-ops/test/mlu_op_gtest/pb_gtest/src/gtest/mlu_op_gtest.cpp:696]: 这个我们需要拷贝出来吗，感觉很奇怪。。。

**Review by DanieeelLiu — APPROVED** [review]
kernels/xxx.cpp kernels/xxx.mlu reviewd

File: docs/bangc-docs/design_docs/focal_loss_sigmoid_forward/focal_loss_sigmoid_forward.md [hunk 4]
```diff
@@ -63,7 +63,7 @@ example:
 -
 ### 1.2 算子功能和应用场景描述
 
-`FocalLoss` 是在`CrossEntropyLoss` 的基础上增加了权重系数`alpha`和聚焦系数`gamma`，其目的是通过减少易分类样本的权重，从而使得模型在训练时更专注于难分类的样本。对于二分类问题：
+`FocalLoss` 是在`CrossEntropyLoss` 的基础上增加了卷积滤波张量系数`alpha`和聚焦系数`gamma`，其目的是通过减少易分类样本的卷积滤波张量，从而使得模型在训练时更专注于难分类的样本。对于二分类问题：
```
Comment (Unireverse) [inline docs/bangc-docs/design_docs/focal_loss_sigmoid_forward/focal_loss_sigmoid_forward.md:66]: 比重

File: docs/bangc-docs/design_docs/ms_deform_attn_forward/ms_deform_attn_forward.md [hunk 5]
```diff
@@ -79,7 +79,7 @@ MSDeformAttn(z_q,\hat p_q,\{ x^l \}_{l=1}^L)=\sum_{m=1}^{M}W_m[\sum_{l=1}^{L}\su
 | data_sampling_loc_desc | 对输入data_sampling_loc的描述                           | 输入              | mluOpTensorDescriptor_t | /        | 无       |
 | data_sampling_loc      | 采样点的归一化坐标                                       | 输入              | float32               | ARRAY    | [bs, num_queries, num_heads, num_levels, num_points, 2]       |
 | data_attn_weight_desc  | 对输入data_attn_weight的描述                            | 输入              | mluOpTensorDescriptor_t | /        | 无       |
-| data_attn_weight       | attention权重值                                         | 输入              | float32               | ARRAY    | [bs, num_queries, num_heads, num_levels, num_points]       |
+| data_attn_weight       | attention卷积滤波张量值                                         | 输入              | float32               | ARRAY    | [bs, num_queries, num_heads, num_levels, num_points]       |
```
Comment (Unireverse) [inline docs/bangc-docs/design_docs/ms_deform_attn_forward/ms_deform_attn_forward.md:82]: 这里感觉替换的也不对

File: bangc-ops/mlu_op.h [hunk 6]
```diff
@@ -1567,6 +1624,65 @@ mluOpSetTensorDescriptorEx(mluOpTensorDescriptor_t desc,
                            const int dimSize[],
                            const int dimStride[]);
 
+// Group:Tensor
+/*!
+ * @brief Initializes the tensor descriptor pointed by \b desc that was previously created
+ * with ::mluOpCreateTensorDescriptor, and sets the information about the
+ * dimensions, strides, data type, and layout of the input tensor.
+ *
+ * Compare with ::mluOpSetTensorDescriptor_v2, you can specify the strides of all dimensions with
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:1633]: Compared with

**Review by tudejiang79 — APPROVED** [review]

**Review by Unireverse — APPROVED** [review]

**Review by GraceGuanH — APPROVED** [review]

**Review by WangYuan-cn — APPROVED** [review]

**Review by ArtIntAI — APPROVED** [review]

**Pull request merged**